generators/app/templates/resources/lib/kodiLogging.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+from resources.lib.utilities import getSettingAsBool
+
+import logging
+import xbmc
+import xbmcaddon
+
+
+class KodiLogHandler(logging.StreamHandler):
+
+    def __init__(self):
+        logging.StreamHandler.__init__(self)
+        addon_id = xbmcaddon.Addon().getAddonInfo('id')
+        prefix = b"[%s] " % addon_id
+        formatter = logging.Formatter(prefix + b'%(name)s: %(message)s')
+        self.setFormatter(formatter)
+
+    def emit(self, record):
+        levels = {
+            logging.CRITICAL: xbmc.LOGFATAL,
+            logging.ERROR: xbmc.LOGERROR,
+            logging.WARNING: xbmc.LOGWARNING,
+            logging.INFO: xbmc.LOGINFO,
+            logging.DEBUG: xbmc.LOGDEBUG,
+            logging.NOTSET: xbmc.LOGNONE,
+        }
+        if getSettingAsBool('debug'):
+            try:
+                xbmc.log(self.format(record), levels[record.levelno])
+            except UnicodeEncodeError:
```

Comment:
perhaps this post may be useful to you:
http://forum.kodi.tv/showthread.php?tid=144677

it explains how to properly handle/avoid unicode erros.

there's an example log function at the bottom of the page, which i've been using in all my addons for years now, and never had an issue with it.